solidity_contracts/src/Kakarot/Coinbase.sol

```diff
@@ -1,23 +1,32 @@
 // SPDX-License-Identifier: MIT
 pragma solidity 0.8.27;
 
-import {DualVmToken} from "../CairoPrecompiles/DualVmToken.sol";
 import {Ownable2Step, Ownable} from "@openzeppelin-contracts/access/Ownable2Step.sol";
+import {CairoLib} from "kakarot-lib/CairoLib.sol";
 
 contract Coinbase is Ownable2Step {
-    /// @dev The EVM address of the DualVmToken for Kakarot ETH.
-    DualVmToken public immutable kakarotEth;
+    using CairoLib for uint256;
+
+    uint256 public immutable nativeTokenStarknetAddress;
 
     /// Constructor sets the owner of the contract
-    constructor(address _kakarotEth) Ownable(msg.sender) {
-        kakarotEth = DualVmToken(_kakarotEth);
+    constructor(uint256 _nativeTokenStarknetAddress) Ownable(msg.sender) {
+        nativeTokenStarknetAddress = _nativeTokenStarknetAddress;
     }
 
-    /// @notice Withdraws ETH from the contract to a Starknet address
-    /// @dev DualVmToken.balanceOf(this) is the same as address(this).balance
+    function receiveEther() external payable {}
+
+    /// @notice Withdraws the native token collected by the contract to an address
+    /// @dev Uses CairoLib to make a StarknetCall to transfer this contract's balance to a starknet address.
     /// @param toStarknetAddress The Starknet address to withdraw to
     function withdraw(uint256 toStarknetAddress) external onlyOwner {
         uint256 balance = address(this).balance;
-        kakarotEth.transfer(toStarknetAddress, balance);
+        uint128 balanceLow = uint128(balance);
```

Comment:
I think that someone reported that solidity is now doing safe casting, meaning that one needs to add `% 2**128`
```suggestion
        uint128 balanceLow = uint128(balance % 2**128);
```

Reply:
After a second look, 
The issue is invalid 
```solidity 
pragma solidity 0.8.27;

import "forge-std/Test.sol";
import "forge-std/console.sol";

contract Ok is Test {
    function test_conversion() public {
        uint256 a = type(uint256).max;
        uint128 b = uint128(a);
        console.log(a);
        console.log(b);
    }
}
```

Logs:
  115792089237316195423570985008687907853269984665640564039457584007913129639935
  340282366920938463463374607431768211455
  
 It compiles and output the value truncaned 
 
 My bad for wrongly doing the verification